File: codility/src/main/java/codility/com/lessons/l10prime/MinPerimeterRectangle.java
package codility.com.lessons.l10prime;


public class MinPerimeterRectangle {
	public int solution(int N) {
		int min = Integer.MAX_VALUE;
		int newValue;
		for (int i = 1; i * i <= N; i++) {
			if (N % i == 0) {
				newValue = 2 * (i + N / i);
				min = min < newValue ? min : newValue;
			}
		}
		return min;
	}
}

File: codility_new/min_perimeter_rectangle.py
# https://app.codility.com/programmers/lessons/10-prime_and_composite_numbers/min_perimeter_rectangle/
import math

def solution(N):
    side = int(math.sqrt(N))
    if side * side == N:
        return 2 * (side + side)
    
    for i in range(side):
        if N % (side - i) == 0:
            a = side - i
            b = N // (side - i)
            # print(f"N: {N}, side: {side}, i: {i}, a: {a}, b: {b}")
            return 2 * (a + b)

    return 0
